docs/config-and-instances.md

```diff
@@ -0,0 +1,340 @@
+  `autoSave: false`. This would lead to a user defined `Config` and a single
+  `VM` instance whenever a an instance is created.
+
+
+## ICID-E - Extended Option “Imperative-Config-Instance-Dance”
+Like Option “Imperative-Config-Instance-Dance” - but with an added initial
+expansion
+
+### Observations
+* Potentially domain modification can be done without runtime
+   * Assumption that the component used for expanding and checking the
+     out-of-runtime spec does comply with the runtime component to run the spec
+     IOW assumption is that libvirt in expansion creates valid expanded domain
+
+
+## Notes
+
+### Presets and flavors
+Presets look unsuited for flavors. The requirements of flavors are to be able
+to set certain defaults for specific hardware, i.e. use specific busses (i.e.
+virtio) or device models (i.e. e1000) by default. But presets only allow to add
+specific branches to the tree (if you look at the specification as a tree).
+Presets do not allow to user certain values as defaults for arbitrary leaves
+in the tree (which is the defaulting to certain values in certain places).
+
+### UI friendly spec modification
+For a UI it would be nice if spec changes (creating, updating [hot-plugging])
+could be done without runtime interactions because of performance reasons.
+
+### Pause and Suspend
+A VM is _paused_ if it's present on a host and consuming memory resources.
+A VM is _suspended_ if it's not present on a host and not consuming memory
+resources.
+In both cases a VM ran be resumed and will continue to work as usual.
+
+Because a VM is present when it's _paused_ it would be a state of an instance.
+And because a VM is not present if it's _suspended_ this should be a state of
+a config. Has to be decided if it's good or bad, but if suspended is a state
+of a config, this would (in the scope of this doc) allow us to create mutliple
+instances from this state. To me it's not a contradiction, we just need to see
+- but this is generally not handled in this document - how to handle the
+storage in such a case.
+
+### Deep snapshots / volume handling
+This document only covers the mechanism on the spec, and how the requirements
+can be achieved on the spec level.
+Additional work is required to fine match this with how this will work on a
+_deep_ scope, thus also snpashotting eventually attached volumes.
+So far it looks like this should be possible.
+
+
+## Discussion
+
+### Overview
+In general the presented approaches could meet the requirements, an overview:
+
+| Requirement        | ICID | ICID-E |
+| Non running VMs    | Config | |
+| Create then launch | Config and instance | |
+| Flavor             | Special crafted Config | |
+| Expanding          | Performed by runtime | |
+| Detailed           | Detailed Config | |
+| Dynamic data       | .state field of instance | |
+| Defaulting         | (Needs improvement) | |
+| Dry-model-for-UI   | - | + |
+	
+
+### Defaulting - Yes, but how to do it better
```

Comment:
Sure, I see #281 as the primary place to discuss defaulting. Droppe this paragraph as well to prevent confusion.